pkg/operator/controller/extension/add.go

```diff
@@ -76,3 +95,16 @@ func (r *Reconciler) MapToAllExtensions(ctx context.Context, log logr.Logger, re
 
 	return mapper.ObjectListToRequests(extensionList)
 }
+
+type disableAnnotationChangedPredicate struct {
+	predicate.Funcs
+}
+
+// Update implements default UpdateEvent filter for validating annotation change.
+func (disableAnnotationChangedPredicate) Update(e event.UpdateEvent) bool {
+	if e.ObjectOld == nil || e.ObjectNew == nil {
+		return false
+	}
+
+	return e.ObjectNew.GetAnnotations()[AnnotationKeyDisableAdmission] != e.ObjectOld.GetAnnotations()[AnnotationKeyDisableAdmission]
+}
```

Comment:
Why do we need this? Can't we just force users of remove the `.spec.deployment.admission` from the API if they don't want deploy the admission component anymore?